Source/Dafny/Dafny.atg

```diff
@@ -4105,13 +4133,22 @@ CasePatternLocal<.out CasePattern<LocalVariable> pat, bool isGhost.>
     ")"                                          (. pat = new CasePattern<LocalVariable>(id, id.val, arguments); .)
   | "("                                          (. id = t;
                                                     arguments = new List<CasePattern<LocalVariable>>();
+                                                    ghostArguments = new List<bool>();
                                                  .)
-      [ CasePatternLocal<out pat, isGhost>       (. arguments.Add(pat); .)
-        { "," CasePatternLocal<out pat, isGhost> (. arguments.Add(pat); .)
+      [
+        [ "ghost"                                        (. ghostArg = true; .)
+        ]
+        CasePatternLocal<out pat, isGhost || ghostArg>   (. arguments.Add(pat); ghostArguments.Add(isGhost || ghostArg); .)
+        { ","                                            (. ghostArg = false; .)
+          [ "ghost"                                      (. ghostArg = true; .)
+          ]
+          CasePatternLocal<out pat, isGhost || ghostArg> (. arguments.Add(pat); ghostArguments.Add(isGhost || ghostArg); .)
         }
       ]
     ")"                                (. // Parse parenthesis without an identifier as a built in tuple type.
-                                          theBuiltIns.TupleType(id, arguments.Count, true); // make sure the tuple type exists
+                                          // TODO(fpoli): Why is the following call needed?
```

Comment:
I'm not sure I understand the question, but `theBuiltIns.TupleType` records which dimensions of tuple types are created in the input program, which causes extra tuple type definitions to be emitted if needed.

Reply:
To expand the question, I'm wondering why the parser needs to emit tuple definitions. I expected the parser to be a component that takes a program and just emits an AST and/or parsing errors, but this doesn't seem the case.

I'll try to comment that out and see what happens, thanks.

Reply:
All tests still pass. I also found somewhere else a comment stating that "the parser should ensure that all needed tuple types exist by the time of resolution", which answers my question.

Reply:
> Remy's observation has some merit. However, I wonder what would happen in an _incorrect_ program that tried something like:
> 
> ```
> var (x, y) := 100;
> ```
> 
> This is not type correct, but it _may_ be that the type checker expects the 2-tuple to exist at this time.

In both cases (before this PR and after commenting the `theBuiltIns.TupleType(..)`) dafny reports the following error. So, it's unclear whether the `theBuiltIns.TupleType(..)` is actually needed.

```
Error: type of RHS (int) does not match type of bound variable '_#Make2'
```